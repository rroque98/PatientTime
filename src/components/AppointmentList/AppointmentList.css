.drHeader {
  font-size: 2em;
  font-weight: 800;
  padding: 1em 0 0 0;
}

.email {
  padding: 0 0 1em 0;
}

table {
  border-collapse: collapse;
  padding: 0.5em 0 0 0;
}

thead {
  background-color: var(--notableBlue);
  color: var(--white);
}

th {
  text-align: left;
  padding: 0.5em;
}

tr:nth-child(even) {
  background: var(--veryLightGrey);
}

td {
  border-left: var(--veryLightGrey) 1px solid;
  border-right: var(--veryLightGrey) 1px solid;
  padding: 0.5em;
}
