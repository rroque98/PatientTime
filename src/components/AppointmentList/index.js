export { default } from "./AppointmentList.jsx";
