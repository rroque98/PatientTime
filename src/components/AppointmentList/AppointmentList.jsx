import React from "react";
import "./AppointmentList.css";

const AppointmentList = ({ appointments }) => {
  let count = 0;
  console.log(appointments);
  return (
    <table>
      <thead>
        <tr>
          <th>#</th>
          <th>Name</th>
          <th>Time</th>
          <th>Kind</th>
        </tr>
      </thead>
      <tbody>
        {appointments.map(appt => {
          count++;
          return (
            <tr key={appt.name}>
              <td>{count}</td>
              <td>{appt.name}</td>
              <td>{appt.time}</td>
              <td>{appt.kind}</td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

export default AppointmentList;
