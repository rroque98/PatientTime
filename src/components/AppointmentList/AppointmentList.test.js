import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "react-testing-library";
import AppointmentList from "./index.js";

afterEach(cleanup);

test("AppointmentList snapshot", () => {
  const appointments = [
    {
      time: "8:15",
      name: "Tameka Facer",
      kind: "New Patient"
    },
    {
      time: "9:15",
      name: "Elma Lewallen",
      kind: "Follow Up"
    },
    {
      time: "11:30",
      name: "Rusty Montufar",
      kind: "New Patient"
    },
    {
      time: "2:15",
      name: "Belma Flinn",
      kind: "New Patient"
    }
  ];
  const apptListComp = render(
    <AppointmentList
      appointments={appointments}
      currPhysFullName={"Rand Phys"}
      currPhysEmail={"randPhys@dr.com"}
    />
  );
  expect(apptListComp).toMatchSnapshot();
});
