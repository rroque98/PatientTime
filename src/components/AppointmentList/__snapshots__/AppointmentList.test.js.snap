// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="drHeader"
      >
        Dr. 
        Rand Phys
      </header>
      <div
        class="email"
      >
        randPhys@dr.com
      </div>
      <table>
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Name
            </th>
            <th>
              Time
            </th>
            <th>
              Kind
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Tameka Facer
            </td>
            <td>
              8:15
            </td>
            <td>
              New Patient
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              Elma Lewallen
            </td>
            <td>
              9:15
            </td>
            <td>
              Follow Up
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Rusty Montufar
            </td>
            <td>
              11:30
            </td>
            <td>
              New Patient
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              Belma Flinn
            </td>
            <td>
              2:15
            </td>
            <td>
              New Patient
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <header
      class="drHeader"
    >
      Dr. 
      Rand Phys
    </header>
    <div
      class="email"
    >
      randPhys@dr.com
    </div>
    <table>
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Name
          </th>
          <th>
            Time
          </th>
          <th>
            Kind
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Tameka Facer
          </td>
          <td>
            8:15
          </td>
          <td>
            New Patient
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Elma Lewallen
          </td>
          <td>
            9:15
          </td>
          <td>
            Follow Up
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            Rusty Montufar
          </td>
          <td>
            11:30
          </td>
          <td>
            New Patient
          </td>
        </tr>
        <tr>
          <td>
            4
          </td>
          <td>
            Belma Flinn
          </td>
          <td>
            2:15
          </td>
          <td>
            New Patient
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
